src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -59,18 +58,10 @@ public function supportsDecoding($format, array $context = []): bool
      *
      * @throws RuntimeException if no decoder is found
      */
-    private function getDecoder(string $format, array $context): DecoderInterface
+    public function getDecoder(string $format, array $context): DecoderInterface
```

Comment:
Why this change? This will make this code harder to maintain. I think that this method should remain private.

Reply:
Because the tests now retrieve/expect a specific encoder/decoder instance.

Reply:
We cannot modify our public API just for a test. This should be tested in another way.
Here you can just register a test encoder returning a known result  ("magic") when properly called.

Reply:
Okay. I can change that - if the general change of this PR is fine for you.